contracts/utils/Whitelist.sol

```diff
@@ -0,0 +1,137 @@
+    modifier onlyWhitelisted() {
+        _onlyWhitelisted();
+        _;
+    }
+
+    /**
+        @notice Enables whitelist feature
+        @dev The caller must have Whitelister or Admin role
+        @dev Enables whitelist if disabled and emits `WhitelistEnabled` event
+     */
+    function enableWhitelist() external onlyWhitelisterOrAdmin {
+        require(!isWhitelistEnabled(), "Whitelist is already enabled");
+        _isWhitelistEnabled = true;
+        emit WhitelistEnabled(msg.sender);
+    }
+
+    /**
+        @notice Disables whitelist feature
+        @dev The caller must have Whitelister or Admin role
+        @dev Disables whitelist if enabled and emits `WhitelistDisabled` event
+     */
+    function disableWhitelist() external onlyWhitelisterOrAdmin {
+        require(isWhitelistEnabled(), "Whitelist is already disabled");
+        _isWhitelistEnabled = false;
+        emit WhitelistDisabled(msg.sender);
+    }
+
+    /**
+        @notice Returns true if whitelist feature is enabled
+     */
+    function isWhitelistEnabled() public view returns (bool) {
+        return _isWhitelistEnabled;
+    }
+
+    /**
+        @notice Adds an address to the whitelist
+        @dev The caller must have Whitelister or Admin role
+        @dev Adds an address if not whitelisted and emits `WhitelistAddressAdded` event
+        @param addressToAdd Address that is added to the whitelist
+     */
+    function addToWhitelist(address addressToAdd) public onlyWhitelisterOrAdmin {
+        require(!isWhitelisted(addressToAdd), "Address to add is already whitelisted");
+        _whitelist.add(addressToAdd);
+        emit WhitelistAddressAdded(addressToAdd);
+    }
+
+    /**
+        @notice Removes an address from the whitelist
+        @dev The caller must have Whitelister or Admin role
+        @dev Removes a whitelisted address and emits `WhitelistAddressAdded` event
+        @param addressToRemove Address that is removed from the whitelist
+     */
+    function removeFromWhitelist(address addressToRemove) public onlyWhitelisterOrAdmin {
+        require(isWhitelisted(addressToRemove), "Address to remove is not whitelisted");
+        _whitelist.remove(addressToRemove);
+        emit WhitelistAddressRemoved(addressToRemove);
+    }
+
+    /**
+        @notice Returns true if address is whitelisted
+        @param addressToCheck Address to check
+     */
+    function isWhitelisted(address addressToCheck) public view returns (bool) {
+        return _whitelist.contains(addressToCheck);
+    }
+
+    // Checks if method caller has Whitelister or Admin role
+    function _onlyWhitelisterOrAdmin() private view {
+        require(hasRole(WHITELISTER_ROLE, msg.sender)
+            || hasRole(DEFAULT_ADMIN_ROLE, msg.sender),
+            "Sender doesn't have Whitelister or Admin role");
+    }
+
+    // Checks if method caller is whitelisted
+    function _onlyWhitelisted() private view {
```

Comment:
Please, add @notice tag here.

Reply:
[added](https://github.com/Cerebellum-Network/chainbridge-solidity/pull/4/files#diff-d1d675d04151a0a3ec00bad5e7e2fbea4c4d2e70a75a00f656df8fa08553f5f7R134) `@dev`, `@notice` is for public ones